joint_calling/upload_processor.py

```diff
@@ -13,13 +13,21 @@
     - User must be authenticated with appropriate permissions """
 
 import os
-from typing import List, Optional
+from typing import List, Optional, NamedTuple
 import hailtop.batch as hb
 
 
+class SampleGroup(NamedTuple):
+    """ Defines a group of files associated with each sample"""
+
+    data_file: str
+    tbi: str
```

Comment:
You're using this for both gVCF and CRAM indexes (`crai`, not `tbi`), so maybe a more generic name is better?

```suggestion
    index: str
```

Reply:
Interestingly, because the `tuple` class has a method `index` we can't do this. But I've switched it to `index_file` which hopefully has the same effect :)